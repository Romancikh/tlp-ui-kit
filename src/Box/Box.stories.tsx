import type { Meta, StoryObj } from "@storybook/react";

import Box from "./Box";

const meta = {
  component: Box,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  title: "Box",
} satisfies Meta<typeof Box>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    border: "1px solid #000",
    children: "Some text",
    height: 200,
    width: 300,
  },
};
