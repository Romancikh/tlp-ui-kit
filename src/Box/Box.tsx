import MuiBox from "@mui/material/Box";
import { PropsWithChildren } from "react";

type BoxProps = PropsWithChildren & {
  alignItems?: "center" | "flex-start";
  bgcolor?: string;
  border?: string;
  borderRadius?: number | string;
  boxSizing?: "border-box";
  display?: "flex";
  flexDirection?: "column";
  gap?: number;
  height?: "fit-content" | number;
  justifyContent?: "flex-start" | "space-between";
  marginLeft?: number;
  marginTop?: number;
  maxWidth?: number;
  padding?: number;
  width?: number | string;
};

function Box({
  alignItems,
  bgcolor,
  border,
  borderRadius,
  boxSizing,
  children,
  display,
  flexDirection,
  gap,
  height,
  justifyContent,
  marginLeft,
  marginTop,
  maxWidth,
  padding,
  width,
}: BoxProps) {
  return (
    <MuiBox
      alignItems={alignItems}
      bgcolor={bgcolor}
      border={border}
      borderRadius={borderRadius}
      boxSizing={boxSizing}
      display={display}
      flexDirection={flexDirection}
      gap={gap}
      height={height}
      justifyContent={justifyContent}
      marginLeft={marginLeft}
      marginTop={marginTop}
      maxWidth={maxWidth}
      padding={padding}
      width={width}
    >
      {children}
    </MuiBox>
  );
}

export default Box;
