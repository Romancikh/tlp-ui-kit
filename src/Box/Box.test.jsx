import { render } from "@testing-library/react";
import Box from "./Box";

test("Box renders correctly", () => {
  const { container } = render(
    <Box
      width={300}
      height={200}
      border="1px solid #000"
      children="Some content"
    />,
  );
  expect(container).toMatchSnapshot();
});
