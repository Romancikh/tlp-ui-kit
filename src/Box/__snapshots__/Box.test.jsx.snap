// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-qhjjpr"
  >
    Some content
  </div>
</div>
`;
